internal/httpserve/handlers/integration_config.go

```diff
@@ -0,0 +1,248 @@
+package handlers
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+
+	echo "github.com/theopenlane/echox"
+	"github.com/xeipuuv/gojsonschema"
+
+	"github.com/theopenlane/iam/auth"
+
+	"github.com/theopenlane/core/internal/httpserve/handlers/internal/jsonschemautil"
+	"github.com/theopenlane/core/internal/integrations/types"
+	credentialmodels "github.com/theopenlane/core/pkg/models"
+	openapi "github.com/theopenlane/core/pkg/openapi"
+	"github.com/theopenlane/utils/rout"
+)
+
+// ConfigureIntegrationProvider stores non-OAuth credentials/configuration for a provider
+func (h *Handler) ConfigureIntegrationProvider(ctx echo.Context, openapiCtx *OpenAPIContext) error {
+	payload, err := BindAndValidateWithAutoRegistry(ctx, h, openapiCtx.Operation, openapi.ExampleIntegrationConfigPayload, openapi.IntegrationConfigResponse{}, openapiCtx.Registry)
+	if err != nil {
+		return h.InvalidInput(ctx, err, openapiCtx)
+	}
+
+	requestCtx := ctx.Request().Context()
+
+	providerKey := payload.Provider
+	if providerKey == "" {
+		return h.BadRequest(ctx, rout.MissingField("provider"), openapiCtx)
+	}
+
+	orgID, err := auth.GetOrganizationIDFromContext(requestCtx)
+	if err != nil {
+		return h.Unauthorized(ctx, err, openapiCtx)
+	}
+
+	if h.IntegrationRegistry == nil {
+		return h.InternalServerError(ctx, errIntegrationRegistryNotConfigured, openapiCtx)
+	}
+
+	providerType := types.ProviderTypeFromString(providerKey)
+	if providerType == types.ProviderUnknown {
+		return h.BadRequest(ctx, ErrInvalidProvider, openapiCtx)
+	}
+
+	spec, ok := h.IntegrationRegistry.Config(providerType)
+	if !ok {
+		return h.BadRequest(ctx, ErrInvalidProvider, openapiCtx)
+	}
+	if !spec.Active {
+		return h.BadRequest(ctx, ErrProviderDisabled, openapiCtx)
+	}
+
+	if len(spec.CredentialsSchema) == 0 {
+		return h.BadRequest(ctx, rout.MissingField("credentialsSchema"), openapiCtx)
+	}
+
+	attrs := payload.Body.ToMap()
+	if len(attrs) == 0 {
+		return h.BadRequest(ctx, rout.MissingField("payload"), openapiCtx)
+	}
+
+	if key, ok := attrs["serviceAccountKey"].(string); ok && strings.TrimSpace(key) != "" {
+		attrs["serviceAccountKey"] = normalizeServiceAccountKey(key)
+	}
+
+	schemaLoader := gojsonschema.NewGoLoader(spec.CredentialsSchema)
+	documentLoader := gojsonschema.NewGoLoader(attrs)
+	result, err := gojsonschema.Validate(schemaLoader, documentLoader)
+	if err != nil {
+		return h.InternalServerError(ctx, err, openapiCtx)
+	}
+
+	if fieldErrs := jsonschemautil.FieldErrorsFromResult(result); len(fieldErrs) > 0 {
+		return h.BadRequest(ctx, fieldErrs[0], openapiCtx)
+	}
+
+	if err := h.persistCredentialConfiguration(requestCtx, orgID, providerType, attrs); err != nil {
+		return h.InternalServerError(ctx, err, openapiCtx)
+	}
+
+	if err := h.runIntegrationHealthCheck(requestCtx, orgID, providerType); err != nil {
```

Comment:
the error should be logged so we have it for debugging
```suggestion
	if err := h.runIntegrationHealthCheck(requestCtx, orgID, providerType); err != nil {
		logx.FromContext(requestCtx).Error().Err(err).Msg("error running integration health check")
```